### DISCUSSION TITLE
How to hide sidebar in blog post?

### DISCUSSION THREAD

**Discussion by sandsoncosta** [hbstack/.github#78] [Q&A] [answered]
I created my blog (https://sandsoncosta.github.io/) and the current layout is fine.
I would like to hide the sidebar on the blog.
How do I do this?
![image](https://github.com/user-attachments/assets/c4fb873c-f54b-4024-a459-f5b06491e7e1)

**Comment by razonyang** [comment]
You'll need to remove the sidebar stuff and then the theme will detect if there is sidebar.

1. Remove sidebar modules, for example, https://github.com/hbstack/theme/blob/67ede1ed1108fca1f048a6824fc98239278f4a4a/config/_default/module.yaml#L27C1-L29C61

2. Remove sidebar hooks, for example, https://github.com/hbstack/theme/blob/67ede1ed1108fca1f048a6824fc98239278f4a4a/config/_default/params.yaml#L113C1-L115

Please restart Hugo server after modifying config.

**Reply by sandsoncosta** [reply]
Yes, I was able to adjust it previously in the way you explained.
But then it removes it completely, right?
Isn't there a way to remove it just from the post and leave it in other places?

**Reply by razonyang** [reply]
I see, currently it's not possible, but it's good feature to have, track this on hbstack/blog#992.

**Comment by razonyang** [comment, marked as answer]
Since [github.com/hbstack/blog@v0.38.0](https://github.com/hbstack/blog/releases/tag/v0.38.0), you can hide the sidebar by setting `sidebar` as `false` on front matter.

> [!NOTE]
> See https://hbstack.dev/modules/overview/#upgrade-modules for upgrading modules.

You can also apply it on all single pages via [cascade](https://gohugo.io/content-management/front-matter/#cascade).

```yaml
// hugo.yaml
cascade:
  - _target:
      kind: page
    params:
      sidebar: false
```

**Comment by sandsoncosta** [comment]
It's works.
Thanks!
